### ISSUE TITLE
How to map HTTP header to gRPC metadata

### ISSUE THREAD

**Issue by nguyenminhthu1994** [envoyproxy/envoy#3038]
Description:
I'm building gRPC-JSON transcoder with Envoy proxy and I want map HTTP header to gRPC metadata but haven't documented on https://www.envoyproxy.io/docs.
I was mapped HTTP header to gRPC metadata by add 'header_name':'data' and read it from 
gRPC metadata on gRPC server.
I want verify my approach is correctly? and can send to me any document about this.
Thank you.

**Comment by mattklein123** [comment]
@nguyenminhthu1994 sorry can't understand the question. Can you rephrase?

**Comment by nguyenminhthu1994** [comment]
@mattklein123 Sorry about indistinct.
I have built gRPC-JSON transcoder with Envoy proxy and next step I want map HTTP header to gRPC metadata but on https://www.envoyproxy.io/docs haven't any document about this.

After, I was added HTTP header is 'metadata-added' with content is 'message'
![request](https://user-images.githubusercontent.com/25716532/38594500-bcaac450-3d70-11e8-9122-e1423f48c687.png)
Finally, I was read 'metadata-added' header from gRPC metadata header and show on console
![result](https://user-images.githubusercontent.com/25716532/38594642-68787f2a-3d71-11e8-8c4b-a3c54c016f6e.png)
according to my understanding, I have mapped HTTP header to gRPC metadata header and Envoy proxy default is supported mapped HTTP header to gRPC metadata header when building gRPC-JSON transcoder.
Can you send to me any document about this?
Thank you.

**Comment by mattklein123** [comment]
@lizan any idea on this one?

**Comment by lizan** [comment]
@nguyenminhthu1994 Yes your understand is correct. gRPC metadata is HTTP header by [its HTTP2 mapping spec](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#requests).

**Comment by nguyenminhthu1994** [comment]
@lizan Thanks for your answer.

**Comment by rutujafirke** [comment]
i want to map http header to grpc metadata using envoy proxy. Can anyone send me the approach for this.
